src/community/clustering.jl

```diff
@@ -156,3 +156,24 @@ function global_clustering_coefficient(g::AbstractGraph)
     ntriangles == 0 && return 1.
     return c / ntriangles
 end
+
+"""
+    triangle_count(g::SimpleGraph)
+    Returns the total number of triangles in a graph
+"""
+function triangle_count(g::AbstractGraph)
+    if !is_directed(g)
+        return Int(round(sum(eigvals(Matrix(adjacency_matrix(g))) .^ 3) / 6))
```

Comment:
I don't think we should use matrices, especially not dense ones. Also calculating eigenvalues can be slow and can lead to numerical errors. So you probably should just adjust adjust your algorithm further below, so that it also works with undirected graphs.

Interesting formula though, where did you find it?

Reply:
yes i tried using trace too (which works on sparse matrices) but it is really slow for sparse and used a lot of memory for dense graphs